Content.Server/Traits/TraitSystem.Functions.cs

```diff
@@ -0,0 +1,280 @@
+    }
+}
+
+/// <summary>
+///     If a trait includes any Psionic Powers, this enters the powers into PsionicSystem to be initialized.
+///     If the lack of logic here seems startling, it's okay. All of the logic necessary for adding Psionics is handled by InitializePsionicPower.
+/// </summary>
+[UsedImplicitly]
+public sealed partial class TraitAddPsionics : TraitFunction
+{
+    [DataField, AlwaysPushInheritance]
+    public List<string>? PsionicPowers { get; private set; } = default!;
+
+    public override void OnPlayerSpawn(EntityUid uid,
+        IComponentFactory factory,
+        IEntityManager entityManager,
+        ISerializationManager serializationManager)
+    {
+        var prototype = IoCManager.Resolve<IPrototypeManager>();
+        var psionic = entityManager.System<PsionicAbilitiesSystem>();
+        if (PsionicPowers is null)
+            return;
+
+        foreach (var powerProto in PsionicPowers)
+            if (prototype.TryIndex<PsionicPowerPrototype>(powerProto, out var psionicPower))
+                psionic.InitializePsionicPower(uid, psionicPower, false);
+    }
+}
+
+/// Handles all modification of Known Languages. Removes languages before adding them.
+[UsedImplicitly]
+public sealed partial class TraitModifyLanguages : TraitFunction
+{
+    /// The list of all Spoken Languages that this trait adds.
+    [DataField, AlwaysPushInheritance]
+    public List<string>? LanguagesSpoken { get; private set; } = default!;
+
+    /// The list of all Understood Languages that this trait adds.
+    [DataField, AlwaysPushInheritance]
+    public List<string>? LanguagesUnderstood { get; private set; } = default!;
+
+    /// The list of all Spoken Languages that this trait removes.
+    [DataField, AlwaysPushInheritance]
+    public List<string>? RemoveLanguagesSpoken { get; private set; } = default!;
+
+    /// The list of all Understood Languages that this trait removes.
+    [DataField, AlwaysPushInheritance]
+    public List<string>? RemoveLanguagesUnderstood { get; private set; } = default!;
+
+    public override void OnPlayerSpawn(EntityUid uid,
+        IComponentFactory factory,
+        IEntityManager entityManager,
+        ISerializationManager serializationManager)
+    {
+        var language = entityManager.System<LanguageSystem>();
+
+        if (RemoveLanguagesSpoken is not null)
+            foreach (var lang in RemoveLanguagesSpoken)
+                language.RemoveLanguage(uid, lang, true, false);
+
+        if (RemoveLanguagesUnderstood is not null)
+            foreach (var lang in RemoveLanguagesUnderstood)
+                language.RemoveLanguage(uid, lang, false, true);
+
+        if (LanguagesSpoken is not null)
+            foreach (var lang in LanguagesSpoken)
+                language.AddLanguage(uid, lang, true, false);
+
+        if (LanguagesUnderstood is not null)
+            foreach (var lang in LanguagesUnderstood)
+                language.AddLanguage(uid, lang, false, true);
```

Comment:
I simply want nullable lists gone here. Make them empty lists instead.
Or AT LEAST add some brackets 
```
if (list is not null)
{
   foreach (var lang in list)
       doSomething()
}
```

Reply:
I'm very keen on flexing my "Line Count To Bracket Ratio" against Wizden, so if I can get away with no brackets, I will lmao.